docs/pages/desktop-access/active-directory-manual.mdx

```diff
@@ -1,486 +1,488 @@
+      $Password=[System.Web.Security.Membership]::GeneratePassword(15,1)
+   } until ($Password -match '\d')
+   $SecureStringPassword=ConvertTo-SecureString $Password -AsPlainText -Force
+   
+   New-ADUser `
+     -Name $Name `
+     -SamAccountName $SamAccountName `
+     -AccountPassword $SecureStringPassword `
+     -Enabled $true
+   ```
+   
+   The password generated for the service account is discarded immediately. 
+   Teleport doesn't need the password because it uses x509 certificates for LDAP 
+   authentication. You can reset the password for the service account if you need 
+   to perform password authentication.
 
-- (!docs/pages/includes/tctl.mdx!)
+1. Set the minimum permissions that must granted to the service account by running the 
+   following script in the PowerShell console:
 
-## Step 1/7. Create a restrictive service account
+   ```powershell
+   # Save your domain's distinguished name to a variable.
+   $DomainDN=$((Get-ADDomain).DistinguishedName)
+   
+   # Create the CDP/Teleport container.
+   # If the command fails with "New-ADObject : An attempt was made to add an object
+   # to the directory with a name that is already in use", it means the object
+   # already exists and you can move on to the next step.
+   New-ADObject -Name "Teleport" -Type "container" -Path "CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN"
+   
+   # Allow Teleport to create LDAP containers in the CDP container.
+   dsacls "CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CC;container;"
+   
+   # Allow Teleport to create and delete cRLDistributionPoint objects in the CDP/Teleport container.
+   dsacls "CN=Teleport,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CCDC;cRLDistributionPoint;"
+   
+   # Allow Teleport to write the certificateRevocationList property in the CDP/Teleport container.
+   dsacls "CN=Teleport,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN " /I:T /G "$($SamAccountName):WP;certificateRevocationList;"
+   
+   # Allow Teleport to create and delete certificationAuthority objects in the NTAuthCertificates container.
+   dsacls "CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):CCDC;certificationAuthority;"
+   
+   # Allow Teleport to write the cACertificate property in the NTAuthCertificates container.
+   dsacls "CN=Public Key Services,CN=Services,CN=Configuration,$DomainDN" /I:T /G "$($SamAccountName):WP;cACertificate;"   
+   ```
 
-Teleport requires a service account to connect to your Active Directory domain.
-We recommend creating a dedicated service account with restrictive permissions
-for maximum security.
+1. Get the security identifier for the new service account.
+   For example, if you used the default service account name, run the following command:
+   
+   ```powershell
+   Get-AdUser -Identity "svc-teleport" | Select SID
```

Comment:
Should we just use `$SamAccountName` here instead of hard coding `svc-teleport` and mentioning "if you used the default"?

Reply:
Oddly enough, I tried to use the `$SamAccountName` in running through the steps and it failed for me every time, so I went brute force with hardcoding the value. I couldn't figure out why it was failing, though. If it legitimately works with envar, I can change it. I just like to keep things empirical :smile:.

Reply:
did you run it in the same session where you ran `$SamAccountName="svc-teleport"`?